submission/submission-security/src/main/kotlin/ebi/ac/uk/security/service/SecurityService.kt

```diff
@@ -4,52 +4,73 @@ import ac.uk.ebi.biostd.persistence.model.User
 import ac.uk.ebi.biostd.persistence.repositories.UserDataRepository
 import ebi.ac.uk.api.security.LoginRequest
 import ebi.ac.uk.api.security.RegisterRequest
-import ebi.ac.uk.security.exception.UserNotFoundException
-import ebi.ac.uk.security.model.UserInfo
-import ebi.ac.uk.security.util.PasswordVerifier
-import ebi.ac.uk.security.util.TokenUtil
-import java.util.UUID
+import ebi.ac.uk.security.integration.SecurityEvents
+import ebi.ac.uk.security.integration.SecurityProperties
+import ebi.ac.uk.security.integration.components.ISecurityService
+import ebi.ac.uk.security.integration.exception.ActKeyNotFoundException
+import ebi.ac.uk.security.integration.exception.LoginException
+import ebi.ac.uk.security.integration.exception.UserAlreadyRegister
+import ebi.ac.uk.security.integration.model.api.UserInfo
+import ebi.ac.uk.security.integration.model.events.UserPreRegister
+import ebi.ac.uk.security.integration.model.events.UserRegister
+import ebi.ac.uk.security.util.SecurityUtil
+import io.reactivex.subjects.Subject
 
-class SecurityService(
+internal class SecurityService(
     private val userRepository: UserDataRepository,
-    private val passwordVerifier: PasswordVerifier,
-    private val tokenUtil: TokenUtil,
-    private val requireActivation: Boolean
-) {
-
-    fun login(loginRequest: LoginRequest): UserInfo {
-        val user = userRepository
-                .findByLoginOrEmail(loginRequest.login, loginRequest.login)
-                .orElseThrow { throw UserNotFoundException(loginRequest.login) }
-
-        if (!passwordVerifier.checkPassword(user.passwordDigest, loginRequest.password)) {
-            throw SecurityException("Given password do not match for user '${loginRequest.login}'")
+    private val securityUtil: SecurityUtil,
+    private val securityProps: SecurityProperties,
+    private val userPreRegister: Subject<UserPreRegister> = SecurityEvents.userPreRegister,
+    private val userRegister: Subject<UserRegister> = SecurityEvents.userRegister
+) : ISecurityService {
+
+    override fun login(loginRequest: LoginRequest): UserInfo =
+        userRepository
+            .findByLoginOrEmailAndActive(loginRequest.login, loginRequest.login, true)
+            .filter { securityUtil.checkPassword(it.passwordDigest, loginRequest.password) }
+            .orElseThrow { LoginException() }
+            .let { UserInfo(securityUtil.createToken(it), it) }
+
+    override fun registerUser(request: RegisterRequest): User {
+        if (userRepository.existsByEmail(request.email)) {
```

Comment:
Why don't you include this case also in the `when` clause?

Reply:
mega pro comment